### DISCUSSION TITLE
Test Data Size

### DISCUSSION THREAD

**Discussion by tystuyfzand** [klauspost/compress#751] [Ideas/Features]
Has there been any thought to forking the test data into another repository, and requiring that to be cloned for testing? It would also allow test files to grow without having to place the burden on users who simply `go get` the package.

I know it's a bit of a weird pattern, but seeing how the zstd data is 23MB, it seems like it'll be a big hit for people with slower connections and CI/CD pipelines that'd have to download it every time. I know it's the reason a lot of people don't use nhooyr/websocket, and it's a completely valid reason.

**Comment by klauspost** [comment]
@tystuyfzand It is a semi-reasonable request, but in the end not a high priority one. It seems more like a tooling problem.

People can make their own silly decisions  - I don't really care much. `nhooyr/websocket` appears dead anyway.

**Reply by tystuyfzand** [reply]
Yeah, unfortunately it seems like you're right about nhooyr/websocket. gorilla/websocket went into archive mode and is no longer being maintained, and I'm not aware of any others that are actively maintained - would be nice to have this available without the test data built into a maintained library, but you're right, it's definitely a potential tooling problem. 

Perhaps Go would be better off adding a way to "ignore" certain files/directories on checkout, if git supports it (which it appears to using sparse-checkout and filter), by using a file like .gitignore for go mod downloads?
